designs/administrator/olm/convey-installing-operator/index.md

```diff
@@ -0,0 +1,42 @@
+---
+parent: Administrator
+version: 4.5
+---
+
+# Convey Operator is Installing
+
+Currently when an operator is installed the user is taken to the **Installed Operators** list view, where it may take some time before the entry for the newly installed operator actually appears, particularly if the operator is installed globally and the user is viewing a namespace the operator will be copied to, and he CSV doesn't yet exist. This design gives the admin user immediate feedback that the operator is installing properly, and can surface errors that may occur during the installation process.
+
+## Installing an Operator
+
+![Install operator](img/1-1-install.png)
+- The user selects the operator they'd like to install and selected installation options (no changes to this screen.)
+
+![Installing operator](img/1-2x-installing-leave.png)
+- The user is now taken to the new installing screen where a spinner conveys the operator is currently being installed.
+- A link is included that allows the user to exit to the Installed Operators list.
+- The user could navigate away from this screen using the left nav, and return to it via the **Installed Operators** list view.
+
+![Installed operator list](img/y-1-list.png)
+- On the Installed Operators list, a row item would always appear (even if the CSV isn’t created in that project yet) that conveys the operator is installing, clicking it would return to the installing screen.
+- Values in columns may be unfilled as neccessary.
+
+![Installing operator](img/1-2-installing.png)
+- The user would return to the installing screen if the operator is still being installed. 
+- If the operator had completed installing (either successfully or not,) the user would just be taken to the operator details using existing behavior.
+- **Definition of 'operator installing':** **status.phase** is not yet **Succeeded**, only when **InstallWaiting** or **AllRequirementsMet**.
```

Comment:
Want to clarify the "_Succeeded_" is from `status.conditions.phase` while "InstallWaiting" and "AllRequirementsMet" are from `status.conditions.reason`. The ideal CSV status transition would be as in **[1]** (see **[2]** for valid CSVphase).

Wondering if we could just mention something similar to:
1. before CSV's `status.conditions.phase` reached to "_Succeeded_"
--> show loading screen
2. if  CSV's `status.conditions.phase` reached to "_Failed_"
--> loading screen errored out

**[1]** ideal CSV status transition for installation:
1. `phase`: _Pending_
      `reason`: RequirementsUnknown

2. `phase`: _InstallReady_
      `reason`: AllRequirementsMet

3. `phase`: _Installing_
      `reason`: InstallWaiting

4. `phase`: _Installing_
      `reason`: InstallSucceeded
--------------------------------[ABOVE THIS LINE: show loading screen]--------
5. `phase`: _Succeeded_
      `reason`: InstallSucceeded 

**[2]** valid CSVphase:
https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/api/apis/operators/clusterserviceversion_types.go#L218

Reply:
Thinking it may be worth exploring ways to expose `status.conditions.reason` + `status.conditions.message` somewhere on the loading screen to show the progress in messages. 
**e.g.**
AllRequirementsMet - all requirements found, attempting install
or
InstallSucceeded - waiting for install components to report healthy

Reply:
Similar concepts apply to "Subscription" and "InstallPlan" objects too. Probably don't need to call out the actual field in the `condition` block. The general idea is the "loading screen" would try exposing messages/status from different objects along the "Operator Installation" process.

Reply:
Thanks for that explanation regarding the statuses Tony, very helpful.

Regarding conveying the statuses on this screen, I was sort of hoping that would happen via the add-on design story that is showing the statuses of all the resources being installed: https://docs.google.com/document/d/1Z0zzAn989uyJNe6UzRQ-y7TVHZ-USWJKVYZ-dcagpRc/edit?usp=sharing